src/registrar/admin.py

```diff
@@ -224,6 +226,31 @@ class ContactAdmin(ListHeaderAdmin):
     search_help_text = "Search by firstname, lastname or email."
 
 
+class DomainApplicationAdminForm(forms.ModelForm):
+    """Custom form to limit transitions to available transitions"""
+
+    class Meta:
+        model = models.DomainApplication
+        fields = "__all__"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        instance = kwargs.get("instance")
+        if instance and instance.pk:
+            current_state = instance.status
```

Comment:
You may want to consider doing some kind of check here on instance.status to handle weird values (None or "") and log it if so ("Status was None"). Don't take my word for it, but I believe status should always have a value... I believe that may be enforced in the model definition but having that kind of redundancy is great for catching bugs sometimes

Reply:
fsm enforces that the status is valid, and so only need to handle when the instance of DomainApplication itself is empty; in that case, the full list of status choices should be displayed, as this is an expected scenario when adding a Domain Application through django admin

Reply:
@dave-kennedy-ecs Ah good point, fair enough